### DISCUSSION TITLE
This library vs Bull

### DISCUSSION THREAD

**Discussion by amitav13** [taskforcesh/bullmq#446] [Q&A] [answered]
Hi, I'm new to Bull/BullMQ. Just starting it off on a new project. Should I be using this library or https://github.com/OptimalBits/bull ? This is supposedly Bull 4?

**Comment by manast** [comment, marked as answer]
BullMQ is a rewrite of Bull in typescript and is where all the new developing is taking place, although we still maintain Bull since the userbase is huge. If you want a more modern api with all the benefits of typescript go for BullMQ, if you want a more mature codebase go for Bull. However, new features would most likely land on BullMQ and not backported to Bull.

**Reply by sibelius** [reply]
can you comment on the differences?

is the rate limiting another logic?

https://docs.bullmq.io/guide/rate-limiting

bullmq is a global rate limit and bull was by queue?

on bullmq if I say 10 jobs at a time, it will only run 10 at a time independent of how many workers I have?

and for bull it will run more than 10 if I have more than 10 workers ?

**Reply by manast** [reply]
There are no such differences, besides you are mixing concurrency and rate-limit. Please check the docs about these two topics. I also have a blog post regarding rate limit: https://blog.taskforce.sh/rate-limit-recipes-in-nodejs-using-bullmq/

**Reply by sibelius** [reply]
If I have 1 or 2 workers the rate limit will keep the same? Half for each worker?

**Reply by manast** [reply]
The rate limit is independent on the amount of workers or the concurrency per worker.
